### ISSUE TITLE
Any chance we can get the auto opt-out of huggingface telemetry?

### ISSUE THREAD

**Issue by 311-code** [lllyasviel/stable-diffusion-webui-forge#1014]
I never know if I'm actually doing this right tbh... But as per [this post](https://old.reddit.com/r/StableDiffusion/comments/1eobdzn/how_is_flux_censored_model_tweaks_or_dataset/lhdopy1/) I personally would love to see some environmental variables added to the webui.bat to automatically protect us forge fans from any telemetry being sent by huggingface-hub in venv/lib/site-packages or any other telemetry that sdforge may not be aware of that could be automatically opted out of. Such as is in transformers/utils/hub.py and hf_api.py. 

Telemetry was also recently an issue in Comfy-cli recently which was sending telemetry data [by default]((https://github.com/Comfy-Org/comfy-cli/commit/c79a1f6e30173c450c847ecd9c8611ef20af3106)) and they made a [ton of removals](https://github.com/Comfy-Org/comfy-cli/commit/7313679fb1bab54119ee41fd5f7ee950b50ac02b) And now it filters properly and seems to better protect users.

The OP misread a line of code about prompt_tracking_consent, but I still don't think their post should have been removed for that because these are legitimate security concerns. I know I'll definitely be renaming all of the dataset with an _ in front now before training just in case ;) But anyway here is some opt out code that could be added if you like

Ps. I never know which lines to put these on or if it's right. Huggingface recently changed TRANSFORMERS_OFFLINE=1 so I just duplicate it a bunch of times in activate.bat in comfyui/venv/scripts/ for example:

```
REM Disable Huggingface Telemetry
set "HF_HUB_DISABLE_TELEMETRY=1"

REM Set Huggingface Transformers to Offline mode 
set "TRANSFORMERS_OFFLINE=1"

Not sure if this next one would cause any issues but I believe it allows from_pretrained to use local_files_only=True universally now I read, it was recently added.

REM Set Huggingface Offline mode
set "HF_HUB_OFFLINE=1"
```

**Comment by rktvr** [comment]
having it on by default and no way to opt out is awful.
anyone know what exactly is being transmitted? i assume prompts/images generated (or info thereof)?

**Comment by 311-code** [comment]
The comfy-cli was on by default but not everyone uses that repo, and if you were lucky it gave a prompt during install and you could say no. I said no when it asked me in the past.

For Huggingface they give the option to disable the telemetry by setting the environmental variables above. The first link explains pretty well what it appears to send. I also agree the hugginggace code could use some better filtering for potentially sensitive data. I did find it interesting that they have trained on albert models on imdb. Kind of want to download it and see whats actually in it.

**Comment by lunar-studio** [comment]
Agreed. But could someone track down their IPs and block them via firewall? I didn’t even realize they had done this.

**Comment by 311-code** [comment]
Yes To get the ip of background apps doing things on your network you can use [WFN repo](https://github.com/wokhan/WFN) to prompt you when something is trying to send out or inbound, it has detailed logging. The 2.6 version seems to work for windows 11, but overall it's a buggy app.  

Then you can set blocks in windows firewall under advanced settings by right clicking inbound and outbound and create a rule.  Then go to custom, all programs, choose the protocol or sleect any, then specify the ip and port ir rnage to block. 

Overall I'd recommend Linux Mint if you care about telemetry though and setting similar firewall prompting when incoming and outgoing connections are made. 

There are so many areas of Windows that expose user data it very difficult to manage and a lot of it tied to pretty essential services in services.msc. for example if you try to block certain ports in svchost.exe from connecting to the internet windows firewall gives warnings about "windows-service hardening rules"  overriding it, which likely means it's Windows is still going to allow outgoing traffic for svdhost.exe traffic for ports you blocked.
